In [ ]:
export OPENAI_API_KEY=':)'

In [ ]:
import os 
from openai import OpenAI


os.environ["OPENAI_API_KEY"] = ":)"
OpenAI.api_key = os.getenv("OPENAI_API_KEY")

client = OpenAI()


batch_input_file = client.files.create(
  file=open("batch_dataset_classification_5K.jsonl", "rb"),
  purpose="batch"
)


In [7]:
batch_input_file_id = batch_input_file.id

client.batches.create(
    input_file_id=batch_input_file_id,
    endpoint="/v1/chat/completions",
    completion_window="24h",
    metadata={
      "description": "dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research."
    }
)

Batch(id='batch_lHHPIm7tJR9SBgIy8adyXPiW', completion_window='24h', created_at=1725113037, endpoint='/v1/chat/completions', input_file_id='file-ANnV8EaMWye4NZbAn7uhUPqa', object='batch', status='validating', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1725199437, failed_at=None, finalizing_at=None, in_progress_at=None, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=0, failed=0, total=0))

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='validating', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=None, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=0, failed=0, total=0))

In [32]:
client.batches.retrieve("batch_mrVWYYvkQTvykelqxFGaeyGy")

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='completed', cancelled_at=None, cancelling_at=None, completed_at=1724813081, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=1724809475, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id='file-taG43Un2WibeJjofHgolhyQ8', request_counts=BatchRequestCounts(completed=48279, failed=0, total=48279))

In [35]:
client.batches.retrieve("batch_lHHPIm7tJR9SBgIy8adyXPiW")

Batch(id='batch_lHHPIm7tJR9SBgIy8adyXPiW', completion_window='24h', created_at=1725113037, endpoint='/v1/chat/completions', input_file_id='file-ANnV8EaMWye4NZbAn7uhUPqa', object='batch', status='completed', cancelled_at=None, cancelling_at=None, completed_at=1725113708, error_file_id=None, errors=None, expired_at=None, expires_at=1725199437, failed_at=None, finalizing_at=1725113507, in_progress_at=1725113041, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id='file-cAQYAsuyGPC79KV1EVRBLBw8', request_counts=BatchRequestCounts(completed=5000, failed=0, total=5000))

In [ ]:
from openai import OpenAI
client = OpenAI()


result = client.files.content("file-cAQYAsuyGPC79KV1EVRBLBw8").content

In [37]:
print(type(result))

<class 'bytes'>


In [38]:
result_file_name = "batch_dataset_classification_output_5K.jsonl"

with open(result_file_name, 'wb') as file:
    file.write(result)

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=473, failed=0, total=48279))  @ 01:08 AM on Wednesday the 28th 

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=956, failed=0, total=48279))   @ 01:09AM on Wednesday the 28th

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=1923, failed=0, total=48279))

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=3367, failed=0, total=48279))

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=3850, failed=0, total=48279))  @ 1:15

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=5775, failed=0, total=48279)) @1:20

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=8675, failed=0, total=48279))  @1:25

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=12039, failed=0, total=48279))  @ 01:32

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=17350, failed=0, total=48279)) @1:42

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=20723, failed=0, total=48279)) @1:48

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=25059, failed=0, total=48279)) @1:57

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=28916, failed=0, total=48279)) @2:05

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=32294, failed=0, total=48279))  @2:11

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=33259, failed=0, total=48279))  @-02:13

Batch(id='batch_mrVWYYvkQTvykelqxFGaeyGy', completion_window='24h', created_at=1724803554, endpoint='/v1/chat/completions', input_file_id='file-Ac1u0TazAnYnfZ3K5xL5LJSA', object='batch', status='in_progress', cancelled_at=None, cancelling_at=None, completed_at=None, error_file_id=None, errors=None, expired_at=None, expires_at=1724889954, failed_at=None, finalizing_at=None, in_progress_at=1724803576, metadata={'description': 'dataset classification of books from project gutenberg to asses which are suitable for use in authorship verification research.'}, output_file_id=None, request_counts=BatchRequestCounts(completed=41934, failed=0, total=48279)) @ 2:33

going to sleep now. meeting the 'Taois tomorrow.